src/PrestaShopBundle/Controller/Admin/Sell/Catalog/MonitoringController.php

```diff
@@ -157,7 +158,10 @@ public function deleteBulkAction(Request $request): RedirectResponse
         $productIds = $this->getBulkProductsFromRequest($request, $gridIdentifiers);
 
         try {
-            $this->getCommandBus()->handle(new BulkDeleteProductCommand($productIds));
+            $this->getCommandBus()->handle(new BulkDeleteProductCommand(
+                $productIds,
+                ShopConstraint::shop($this->getContextShopId())
```

Comment:
Are you sure it shouldn't be all shops constraint? I'm really not sure I'm just asking

Reply:
hmm 
as I see monitoring grid queries are based on context shop id, so Im pretty sure it should also delete product only for context shop. Altough I see that monitoring page also works in group and all shops context (but the group context is not handled in grid query builders - so I guess there is already an issue that nobody yet reported). 
So ideally (at least until we get rid of context and add query params handling) it could probably delete by shop constraint based on context (shop, group or allShops), but as I said grid query is not handling group context, so probably it might just be safer to remove product only for context shop id).
Another option would be for now to support deletion for all shops and single shop (so when its group context it would still delete only from context shop, but when its all shops, it would delete from all shops :man_shrugging: ). Tell me what you think

Reply:
Hmm, if it's not clear it's probably safer to only remove from one shop. It's preferable to removing too much

Reply:
Although @MatShir should probably look into this use case, chances are that it's already not handled ideally right now